components/Seo.php

```diff
@@ -74,7 +74,13 @@ private function getDataFromBlogPost()
             $this->title = $post->title;
             $this->description = $post->excerpt;
             $this->keywords = $post->seo_keywords;
-            $this->canonicalUrl = $post->seo_canonical_url ?? Request::url();
+            // The seo_canonical_url saves empty anyway so we need to check 
+            // for it instead of relying on its (non)existence
+            if ( !empty($post->seo_canonical_url) ) {
+                $this->canonicalUrl = $post->seo_canonical_url;
+            } else {
+                $this->canonicalUrl = Request::url();
+            }            
```

Comment:
I like to have whitespace when code block has to do some logic different than the statements next to it, so add newline before `// The seo_canonical_url saves empty anyway so we need to check ` and after closing `else` bracket

or move the if after `$this->robotsFollow = $post->seo_robots_follow;` with newline